### ISSUE TITLE
support "I'd" as ambiguous contraction

### ISSUE THREAD

**Issue by benhjames** [nlp-compromise/compromise#352]
For example, in these cases, `I would` should be `I had`.

```
nlp("I'd better go to the supermarket.").normalize().out('text');
// I would better go to the supermarket.

nlp("I'd wondered why that was.").normalize().out('text');
// I would wondered why that was.

nlp("I'd often contemplated Euler's theorems.").normalize().out('text');
// I would often contemplated euler's theorems.
```

This might be quite tricky to do though as it relies on the tense of the following clause. If the third example had been instead `I'd often contemplate Euler's theorems` i.e. `contemplate` instead of `contemplated`, then `I would ...` makes perfect sense!

**Comment by spencermountain** [comment]
ah! interesting. yeah, we can do this. good idea.
yeah, do you think it's as simple as looking if the next verb is past-tense?
* i'd contemplate -> i would contemplate
* i'd contemplated -> i had contemplated
?

**Comment by benhjames** [comment]
Yeah, I think generally that works with the exception of the special case `I'd better go` => `had` despite not being past tense!

(btw this also applies to other pronouns `you`, `she`, `he` etc)

**Comment by spencermountain** [comment]
hey, this is implemented in v9.0.0:
```js
let r = nlp('i\'d contemplate');
console.log(r.normalize().out('text'));

r = nlp('i\'d contemplated');
console.log(r.normalize().out('text'));

r = nlp('i\'d really contemplated');
console.log(r.normalize().out('text'));
// i would contemplate
// i had contemplated
// i had really contemplated
```
thanks ben!

**Comment by benhjames** [comment]
Amazing! 🎉
